src/sentry/event_manager.py

```diff
@@ -2163,12 +2160,37 @@ def _process_existing_aggregate(
 )
 
 
-def _get_severity_metadata_for_group(event: Event) -> Mapping[str, Any]:
+def _get_severity_metadata_for_group(
+    event: Event, project_id: int, group_type: int | None
+) -> Mapping[str, Any]:
     """
-    Returns severity metadata for an event if the feature flag is enabled.
-    Returns {} on feature flag not enabled or exception.
+    Returns severity metadata for an event if all of the following are true
+    - the feature flag is enabled
+    - the event platform supports severity
+    - the event group type is an error
+
+    Returns {} if conditions aren't met or on exception.
     """
-    if features.has("projects:first-event-severity-calculation", event.project):
+    from sentry.receivers.rules import PLATFORMS_WITH_PRIORITY_ALERTS
+
+    is_supported_platform = any(
+        event.platform.startswith(platform) for platform in PLATFORMS_WITH_PRIORITY_ALERTS
+    )
+    is_error_group = group_type == ErrorGroupType.type_id if group_type else True
+    feature_enabled = features.has("projects:first-event-severity-calculation", event.project)
+
+    should_calculate_severity = feature_enabled and is_supported_platform and is_error_group
+    if should_calculate_severity:
+        from sentry import ratelimits as ratelimiter
+
+        if ratelimiter.backend.is_limited(
+            f"seer:severity-calculation:{project_id}",
+            limit=5,
```

Comment:
just a suggestion - you could make this an option so you can adjust it quickly once it's in production